doc/dev/adapters.md

```diff
@@ -1,15 +1,699 @@
+  ([instances](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#instances)), where it can be sent
+  ([handlers](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#handlers)), and when to send it
+  ([rules](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#rules)).
+
+![operator, adapter and template devs](./img/operator%20template%20adapter%20dev.svg)
+
+Now let's look more closely at how adapters work.
+
+# Template overview
+
+To understand how an adapter instance receives and processes a *template's* specific instances, this section first
+provides details about various artifacts of a *template* that are relevant for adapter development.
+
+As we saw in the previous section, a build of Mixer supports a set of *templates*, and every *template* defines a kind
+of data Mixer dispatches to adapters when processing a request, and also defines the interface for adapters to consume
+that data. The kind of data is expressed as a Go struct and the interface adapters implement is expressed as a Go interface.
+
+The following diagram shows the various components of a template.
+
+![template generated artifacts](./img/template%20generated%20artifacts.svg)
+
+We'll look at each of these in more detail below.
+
+## Template proto file
+
+Templates are defined using a proto file with a message named `'Template'`. `Template` is a simple proto message with
+no associated code. All of the Go artifacts used by adapters are code generated from the template protos.
+
+Every template also has two additional properties associated with it:
+
+* **Name**: Every template has a unique name. Adapter code uses the name of the template to register with Mixer that it
+wants to consume `Instance` objects associated with a particular template. The template name is also used within
+operator config to provide template-specific fields to attribute mapping, which is used to create `Instance `objects.
+
+* **Template_variety**: Every template has a specific `template_variety` which can be either Check, Report or Quota.
+The template and its variety determine the signatures of the methods the adapter must implement for consuming the
+associated instances. The `template_variety` also determines under which of the core Mixer behaviors, check report or
+quota, the `instances` for the templates should be created and dispatched to adapters.
+
+## Generated Go code
+
+Individual templates are processed in order to produce four Go artifacts.
+
+### Instance struct
+
+The `Instance` struct defines the data that is passed to the adapters at request time. During request time, Mixer
+constructs objects of the `Instance` type, based on the request attributes and operator configuration.
+
+### Handler interface
+
+The `Handler` interface defines methods that Mixer uses to dispatch created `Instance` objects to the adapters during
+request time. Adapters implement per template `Handler` interface for Mixer to call into them.
```

Comment:
I suggest replacing the second sentence with:

Adapters must implement one Handler interface per support template type.